std-bits/table/src/main/java/org/enso/table/data/column/storage/numeric/AbstractLongStorage.java

```diff
@@ -18,17 +20,60 @@
 import org.enso.table.data.column.operation.map.numeric.comparisons.LessComparison;
 import org.enso.table.data.column.operation.map.numeric.comparisons.LessOrEqualComparison;
 import org.enso.table.data.column.operation.map.numeric.isin.LongIsInOp;
-import org.enso.table.data.column.storage.*;
+import org.enso.table.data.column.storage.BoolStorage;
+import org.enso.table.data.column.storage.ColumnLongStorage;
+import org.enso.table.data.column.storage.ColumnStorageWithNothingMap;
+import org.enso.table.data.column.storage.Storage;
+import org.enso.table.data.column.storage.ValueIsNothingException;
 import org.enso.table.data.column.storage.type.IntegerType;
 import org.enso.table.data.column.storage.type.StorageType;
+import org.enso.table.data.mask.OrderMask;
+import org.enso.table.data.mask.SliceRange;
+import org.enso.table.problems.BlackholeProblemAggregator;
 import org.graalvm.polyglot.Context;
 
-public abstract class AbstractLongStorage extends NumericStorage<Long>
+public abstract class AbstractLongStorage extends Storage<Long>
     implements ColumnLongStorage, ColumnStorageWithNothingMap {
-  public abstract long getItem(int idx);
-
   private static final MapOperationStorage<Long, AbstractLongStorage> ops = buildOps();
 
+  private final long size;
+  private final IntegerType type;
+  protected BitSet isNothing;
+
+  protected AbstractLongStorage(long size, IntegerType type, BitSet isNothing) {
```

Comment:
I'm surprised `BitSet` is part of the `AbstractLongStorage`.

Passing `null` bitset in `ComputedNullableLongStorage` seems a bit off. Shouldn't we have separate implementations that have and don't have the bitset?